toolchain/check/import_ref.cpp

```diff
@@ -2016,11 +1874,16 @@ static auto TryResolveTypedInst(ImportRefResolver& resolver,
   }
   auto parent_scope_id =
       GetLocalNameScopeId(resolver, import_function.parent_scope_id);
-  LoadLocalPatternConstantIds(resolver,
-                              import_function.implicit_param_patterns_id);
-  LoadLocalPatternConstantIds(resolver, import_function.param_patterns_id);
+  auto implicit_param_patterns = GetLocalInstBlockContents(
+      resolver, import_function.implicit_param_patterns_id);
+  auto param_patterns =
+      GetLocalInstBlockContents(resolver, import_function.param_patterns_id);
   auto generic_data = GetLocalGenericData(resolver, import_function.generic_id);
   auto& new_function = resolver.local_functions().Get(function_id);
```

Comment:
Consider moving `new_function` lower, to just before the `HasNewWork` call. I think it's actually OK here, but my worry is that `GetLocalConstantInstId` sounds like it might be modifying the local SemIR, and might somehow take some action that invalidates the reference. (Also it's a little clearer that it's only needed in the retry and phase 3 work if it's moved later.)